signac/core/jsondict.py

```diff
@@ -194,8 +194,41 @@ def buffer_reads_writes(buffer_size=DEFAULT_BUFFER_SIZE, force_write=False):
 
 
 class JSONDict(SyncedAttrDict):
-
-    def __init__(self, parent=None, filename=None, write_concern=False):
+    """A dict-like mapping interface to a persistent JSON-file.
+
+    The JSONDict is a :class:`~collections.abc.MutableMapping` and therefore
+    behaves similar to a :class:`dict`, but all data is stored persistently in
+    the associated JSON-file on disk.
+
+    .. code-block:: python
+
+            doc = JSONDict('data.json', write_concern=True)
+            doc['foo'] = "bar"
+            assert doc.foo == doc['foo'] == "bar"
+            assert 'foo' in doc
+            del doc['foo']
+
+    This class allows access to keys both with the slicing and attributes.
```

Comment:
This class allows access to values through key indexing or attributes named by keys.

A couple reasons:
1. You're not accessing keys, you're accessing values.
2. Slicing is specific to list-like objects where `[start:stop:step]` syntax applies.

Reply:
Thank you for helping me with that formulation. I've been struggling with that for a while now.